### DISCUSSION TITLE
Bisect script

### DISCUSSION THREAD

**Discussion by Kixunil** [rust-bitcoin/rust-bitcoin#2956] [General]
Just sharing some knowledge. I've found that the only sane way to bisect the `bitcoin` crate if you need to test the API is to make a new binary crate in the workspace and patch `Cargo.toml` in the script. Assuming you have a crate called `bitcoin-bisect` (`cargo new --bin bitcoin-bisect`) this script will handle bisecting for you:

```sh
#!/bin/bash

sed -i.b '/^members =/s/\]$/, "bitcoin-bisect"]/' Cargo.toml || exit 255
cargo +stable run -p bitcoin-bisect
ret=$?
mv Cargo.toml.b Cargo.toml || exit 255
exit $ret
```

**Comment by tcharding** [comment]
Nice, I usually use a crate in another temporary directory with the manifest pointing to the path to the `rust-bitcoin` worktree I'm bisecting. (But I have to remember to run `cargo update` all the time, so your solution seems better.)

**Reply by Kixunil** [reply]
I tried to do that but was getting horrible breakages because of our `[patch]`es. Maybe there's a different trick to solve this but what I found seems simple and it works.
